### ISSUE TITLE
Upgrade from 1.24.x to 1.25.x doesn't replace kube-apiserver...

### ISSUE THREAD

**Issue by seimic** [rancher/rke2#4221]
**Environmental Info:**
RKE2 Version: v1.25.9+rke2r1 (842d05e64bcbf78552f1db0b32700b8faea403a0)

Node(s) CPU architecture, OS, and Version: x86_64 (Ubuntu 22.04.2 LTS)

Cluster Configuration: 3 servers, 3 agents

**Describe the bug:**
After upgrade from 1.24.x to 1.25.x using the system-upgrade-controller some of the deployments remain untouched as it seams.
kube-apiserver, kube-controller-manager and kube-scheduler are still 1.24.x (rancher/hardened-kubernetes:v1.24.13-rke2r1-build20230412)

**Steps To Reproduce:**

1. Install Version 1.24.13+rke2r1
curl -sfL https://get.rke2.io | sudo INSTALL_RKE2_VERSION="v1.24.13+rke2r1" sh -
Use "profile: cis-1.6" in config and start the server.
2. Prepare the configuration as mentioned here: https://docs.rke2.io/known_issues#hardened-125
3. Install system-upgrade-controller and apply an upgrade plan to version 1.25.x (https://docs.rke2.io/upgrade/automated_upgrade)

**Expected behavior:**
All Pods with new version

**Actual behavior:**
Everything seems to work correctly but kube-apiserver, kube-controller-manager and kube-scheduler are still v1.24.x

**Additional context / logs:**
Manual upgrade, installing RKE2 1.25.x was exactly the same.
Did I miss a step in the upgrade procedure? I don't see which.

**Comment by brandond** [comment]
Please check the actual static pods manifest on disk, and the version reported in the pod logs. There is an upstream Kubernetes issue with static pods where the mirror pods visible via kubectl can be out of sync with the actual running state.

**Comment by seimic** [comment]
Hi,

all server nodes have this image in the manifests (in /var/lib/rancher/rke2/agent/pod-manifests): "index.docker.io/rancher/hardened-kubernetes:v1.25.9-rke2r1-build20230412"
What to do to apply this manifests? 'touch' on manifest files? They have the timestamp of the
upgrade when I see it right. Any explosions to expect by doing this? 

Kind regards,
Michael

**Comment by brandond** [comment]
Yep, so everything's fine. The kubelet is just not updating the mirror pods for whatever reason.
